cpp/src/io/comp/debrotli.cu

```diff
@@ -1377,12 +1377,12 @@ static __device__ debrotli_huff_tree_group_s *HuffmanTreeGroupInit(debrotli_stat
                                                                    uint32_t max_symbol,
                                                                    uint32_t ntrees)
 {
-  debrotli_huff_tree_group_s *group = (debrotli_huff_tree_group_s *)local_alloc(
-    s, sizeof(debrotli_huff_tree_group_s) + ntrees * sizeof(uint16_t *) - sizeof(uint16_t *));
-  group->alphabet_size = (uint16_t)alphabet_size;
-  group->max_symbol    = (uint16_t)max_symbol;
-  group->num_htrees    = (uint16_t)ntrees;
-  group->htrees[0]     = nullptr;
+  debrotli_huff_tree_group_s *group = reinterpret_cast<debrotli_huff_tree_group_s *>(local_alloc(
+    s, sizeof(debrotli_huff_tree_group_s) + ntrees * sizeof(uint16_t *) - sizeof(uint16_t *)));
+  group->alphabet_size              = (uint16_t)alphabet_size;
+  group->max_symbol                 = (uint16_t)max_symbol;
+  group->num_htrees                 = (uint16_t)ntrees;
```

Comment:
What about these  c-style casts?  I wonder why it was considered safe to convert  32-bit to 16-bit here.

Reply:
I'm just as curious about the motive. There are still a number of c-style casts within `cudf` that need fixing.
This PR addresses the c-style pointer casts in `cudf::io`. If time permits, I'll try to address the c-style value casts too in another PR.